addons/mail/static/src/models/message_context_menu.js

```diff
@@ -0,0 +1,68 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { one, many, attr } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'MessageContextMenu',
+    recordMethods: {
+
+        
+        /**
+         * Returns whether the given html element is inside this message context menu.
+         *
+         * @param {Element} element
+         * @returns {boolean}
+         */
+        containsElement(element) {
+            return Boolean(this.component && this.component.root.el && this.component.root.el.firstChild.contains(element));
+        },
+    },
+    fields: {
+        messageContextMenuDialog: one('Dialog', {
+            identifying: true,
+            inverse: 'messageContextMenu',
+        }),
+        messageViewOwner: one('MessageView', {
+            related: 'messageContextMenuDialog.messageViewOwnerAsContextMenu'
+        }),
+        messageContextReactionItems: many('MessageReactionGroupItem', {
+            compute() {
+                if (this.messageViewOwner.message.messageReactionGroups.length === 0) {
+                    return clear();
+                }
+                return this.messageViewOwner.message.messageReactionGroups.map(messageReactionGroup => {
+                     return { messageReactionGroup: messageReactionGroup };
+                });
+            },
+            inverse: 'messageContextViewOwner',
+        }),
+        messageContextReactionPartners: many('MessageReactionGroupPartner',{
+            compute() {
+                if (this.messageViewOwner.message.messageReactionGroups.length === 0) {
+                    return clear();
+                }
+                return this.messageViewOwner.message.messageReactionGroups.map(messageReactionGroup => {
+                     return { messageReactionGroup: messageReactionGroup };
+                });
+            },
+            inverse: 'messageContextViewOwner',
+        }),
```

Comment:
Can be renamed to just `personaViews`.
Not entirely sure it's gonna be final name. However, this will improve readability of code, and therefore help choosing better naming (or make proper other modelling changes)